File: 251001/Homeliyanskaya/user-stories/src/main/java/com/homel/user_stories/mapper/LabelMapper.java
package com.homel.user_stories.mapper;

import com.homel.user_stories.dto.LabelRequestTo;
import com.homel.user_stories.dto.LabelResponseTo;
import com.homel.user_stories.model.Label;
import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

@Mapper
public interface LabelMapper {
    LabelMapper INSTANCE = Mappers.getMapper(LabelMapper.class);

    Label toEntity(LabelRequestTo dto);
    LabelResponseTo toResponse(Label entity);
}

File: 251003/Pankratiev/rest/publisher/src/main/kotlin/bsuir/dc/publisher/mapper/LabelMapper.kt
package bsuir.dc.publisher.mapper

import bsuir.dc.publisher.dto.from.LabelRequestTo
import bsuir.dc.publisher.dto.to.LabelResponseTo
import bsuir.dc.publisher.entity.Label

fun LabelRequestTo.toEntity() = Label(
    id = this.id,
    name = this.name
)

fun Label.toResponse() = LabelResponseTo(
    id = this.id,
    name = this.name
)
